### ISSUE TITLE
Settings cleanup

### ISSUE THREAD

**Issue by milesfrain** [nwolverson/vscode-ide-purescript#151]
Some settings have questionable defaults or other issues. Cleaning this up will make configuration a lot easier for beginners.

**purescript.addSpagoSources**
This should default to true.

**purescript.buildCommand:**
```
pulp build -- --json-errors
```
Should this default to a `spago` option? I've just set this field to empty.

**purescript.editorMode**
This setting should be removed.
```
purs ide server -p 15847 --editor-mode src/**/*.purs 
The --editor-mode flag is deprecated and ignored. It's now the default behaviour and the flag will be removed in a future version
```

**purescript.outputDirectory**
Change default from `null` to `""`. See #145 

**purescript.packagePath**
The default of `bower_components` is confusing.
I believe this is fine in practice for `spago` projects with `addSpagoSources`, since this extra path is harmless. But it would be clearer to create another setting, such as `addBowerSources` which adds `boter_components` path, and then convert this setting to something like `additionalPackagePaths` that's blank by default.

**purescript.pscIdeServerExe**
```
psc-ide-server
```
Do we need this anymore? It's been [deprecated](https://github.com/kRITZCREEK/psc-ide) for the past 4 years.

**purescript.useCombinedExe**
> Whether to use the new combined purs executable. This will default to true in the future then go away.

Is it time for this to "go away"?

**Comment by nwolverson** [comment]
I've intended to make a single update that changes/removes deprecated things all at once, evidently it has been a long time coming. However the editor mode flag change is only new in PureScript 0.13.8, so it's not appropriate to remove that option yet - instead that/related can be deprecated and defaulted to the "new world" for the time being.

Otherwise I agree with most of this. I think I'd go as far as to just require adding `bower_component` as manual glob override at this point

**Comment by nwolverson** [comment]
This also duplicates #144

**Comment by nwolverson** [comment]
Implemented in 0.21.2
